junit-platform-launcher/src/main/java/org/junit/platform/launcher/listeners/LoggingListener.java

```diff
@@ -79,14 +79,15 @@ public static LoggingListener forJavaUtilLogging(Level logLevel) {
 	 * @see #forJavaUtilLogging()
 	 * @see #forJavaUtilLogging(Level)
 	 */
-	public static LoggingListener forBiConsumer(BiConsumer<Throwable, Supplier<String>> logger) {
+	public static LoggingListener forBiConsumer(BiConsumer<@Nullable Throwable, Supplier<String>> logger) {
```

Comment:
In most of the cases, the proper signature for `Function`-receiving methods is as follows:

```suggestion
	public static LoggingListener forBiConsumer(BiConsumer<? super @Nullable Throwable, ? super Supplier<String>> logger) {
```

That would allow users to pass even something that `implements BiConsumer<@Nullable Object, @Nullable Object>`.

Unfortunately, [Java uses use-site variance](https://gist.github.com/vlsi/3e7c6a0aa3f13ec0cfc94c0f7447ceb7#issue-1-filter-does-not-accept-a-more-generic-predicate), so effectively it requires users to add `super` and `extends` for almost every method that uses generic parameters.